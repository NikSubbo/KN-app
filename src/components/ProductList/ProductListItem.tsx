import { FC } from 'react';
import TagList from '../TagList/TagList';

type Props = {
  name: string;
  category: string;
  tags: string[];
  handleProductClick: (name: string) => void;
};

export const ProductListItem: FC<Props> = ({
  name,
  category,
  tags,
  handleProductClick
}) => {
  const onClick = () => {
    handleProductClick(name);
  };

  return (
    <button
      className="flex justify-between bg-white rounded shadow-sm py-3.5 px-6"
      onClick={onClick}
    >
      <div>
        <h3 className="text-sm font-medium text-black text-left">{name}</h3>
        <TagList className="mt-1" tags={tags} />
      </div>
      <p className="flex self-center text-darkGray text-sm whitespace-nowrap ml-4">
        {category}
      </p>
    </button>
  );
};

export default ProductListItem;
