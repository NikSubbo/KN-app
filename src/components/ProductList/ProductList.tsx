import { FC } from 'react';
import { Product } from '../../utils/utils';
import ProductListItem from './ProductListItem';

type Props = {
  products: Product[];
  handleProductClick: (name: string) => void;
};

export const ProductList: FC<Props> = ({ products, handleProductClick }) => {
  return (
    <div className="flex flex-col space-y-2">
      {products.map(({ productName, tags, category }) => {
        return (
          <ProductListItem
            key={productName}
            name={productName}
            tags={tags}
            category={category}
            handleProductClick={handleProductClick}
          />
        );
      })}
    </div>
  );
};

export default ProductList;
