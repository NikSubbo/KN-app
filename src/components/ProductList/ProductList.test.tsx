import { render } from '@testing-library/react';
import ProductList from './ProductList';

const DEFAULT_PROPS = {
  products: [
    {
      productName: 'Foxit software PhantomPDF Standard',
      tags: ['PDF', 'Change', 'Create', 'Maintenance', 'Business', 'FoxIT'],
      category: 'Text Editors',
      manufacturerUrl: 'https://www.foxitsoftware.com/de/pdf-editor',
      description: [
        'PhantomPDF provides powerful PDF Editor capabilities to allow authors to update their documents themselves.',
        'Standard - Simple interface and limited functionality.'
      ],
      option1: '1 Year Maintenance',
      option2: 'Without Maintenance'
    },
    {
      productName: 'Foxit Software PhantomPDF Business',
      tags: [],
      category: 'Text Editors',
      manufacturerUrl: 'https://www.foxitsoftware.com/de/pdf-editor',
      description: [
        'PhantomPDF provides powerful PDF Editor capabilities to allow authors to update their documents themselves.',
        'Business - Advances capabilities, complete lifecycle solution.'
      ],
      option1: null,
      option2: null
    }
  ],
  handleProductClick: jest.fn()
};

describe('ProductList', () => {
  it('should match the snapshot', () => {
    const { container } = render(<ProductList {...DEFAULT_PROPS} />);
    expect(container).toMatchSnapshot();
  });
});
