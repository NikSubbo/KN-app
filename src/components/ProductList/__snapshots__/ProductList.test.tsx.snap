// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList should match the snapshot 1`] = `
<div>
  <div
    class="flex flex-col space-y-2"
  >
    <button
      class="flex justify-between bg-white rounded shadow-sm py-3.5 px-6"
    >
      <div>
        <h3
          class="text-sm font-medium text-black text-left"
        >
          Foxit software PhantomPDF Standard
        </h3>
        <div
          class="mt-1 flex flex-wrap"
        >
          <div
            class="text-xs font-medium text-aqua px-2 py-1 rounded bg-aquaLight mr-2 mt-2"
          >
            PDF
          </div>
          <div
            class="text-xs font-medium text-aqua px-2 py-1 rounded bg-aquaLight mr-2 mt-2"
          >
            Change
          </div>
          <div
            class="text-xs font-medium text-aqua px-2 py-1 rounded bg-aquaLight mr-2 mt-2"
          >
            Create
          </div>
          <div
            class="text-xs font-medium text-aqua px-2 py-1 rounded bg-aquaLight mr-2 mt-2"
          >
            Maintenance
          </div>
          <div
            class="text-xs font-medium text-aqua px-2 py-1 rounded bg-aquaLight mr-2 mt-2"
          >
            Business
          </div>
          <div
            class="text-xs font-medium text-aqua px-2 py-1 rounded bg-aquaLight mr-2 mt-2"
          >
            FoxIT
          </div>
        </div>
      </div>
      <p
        class="flex self-center text-darkGray text-sm whitespace-nowrap ml-4"
      >
        Text Editors
      </p>
    </button>
    <button
      class="flex justify-between bg-white rounded shadow-sm py-3.5 px-6"
    >
      <div>
        <h3
          class="text-sm font-medium text-black text-left"
        >
          Foxit Software PhantomPDF Business
        </h3>
      </div>
      <p
        class="flex self-center text-darkGray text-sm whitespace-nowrap ml-4"
      >
        Text Editors
      </p>
    </button>
  </div>
</div>
`;
